src/platform_linux/platform.c

```diff
@@ -56,45 +56,46 @@ void
 platform_heap_destroy(platform_heap_handle UNUSED_PARAM(*heap_handle))
 {}
 
-buffer_handle *
-platform_buffer_create(size_t               length,
-                       platform_heap_handle UNUSED_PARAM(heap_handle),
-                       platform_module_id   UNUSED_PARAM(module_id))
+/*
+ * platform_buffer_init() - Initialize an input buffer_handle, bh.
+ *
+ * Certain modules, e.g. the buffer cache, need a very large buffer which
+ * may not be serviceable by the heap. Create the requested buffer using
+ * mmap() and initialize the input 'bh' to track this memory allocation.
+ */
+platform_status
+platform_buffer_init(buffer_handle *bh, size_t length)
 {
-   buffer_handle *bh = TYPED_MALLOC(platform_get_heap_id(), bh);
+   platform_status rc = STATUS_NO_MEMORY;
 
-   if (bh != NULL) {
-      int prot  = PROT_READ | PROT_WRITE;
-      int flags = MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE;
-      if (platform_use_hugetlb) {
-         flags |= MAP_HUGETLB;
-      }
+   int prot  = PROT_READ | PROT_WRITE;
+   int flags = MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE;
+   if (platform_use_hugetlb) {
+      flags |= MAP_HUGETLB;
+   }
 
-      bh->addr = mmap(NULL, length, prot, flags, -1, 0);
-      if (bh->addr == MAP_FAILED) {
-         platform_error_log(
-            "mmap (%lu) failed with error: %s\n", length, strerror(errno));
-         goto error;
-      }
+   bh->addr = mmap(NULL, length, prot, flags, -1, 0);
+   if (bh->addr == MAP_FAILED) {
+      platform_error_log(
+         "mmap (%lu bytes) failed with error: %s\n", length, strerror(errno));
+      goto error;
+   }
 
-      if (platform_use_mlock) {
-         int rc = mlock(bh->addr, length);
-         if (rc != 0) {
-            platform_error_log(
-               "mlock (%lu) failed with error: %s\n", length, strerror(errno));
-            munmap(bh->addr, length);
-            goto error;
-         }
+   if (platform_use_mlock) {
+      int rc = mlock(bh->addr, length);
```

Comment:
Not you fault but...

I just noticed that this `rc` shadows the `platform_status rc` declared on line 69.

Also, when the `mlock` fails, we return `STATUS_NO_MEMORY`.

Could you change the name of this `rc` so it doesn't shadow the previous, and set a better error code for our return in this case?  Maybe `STATUS_NO_PERMISSION`?

Reply:
Oh or maybe `CONST_STATUS(errno)`?

Reply:
Renamed `rc` on L85 to `mlock_rv` and on error, will set `rc` to `CONST_STATUS(errno)`.